include/SFML/Window/Keyboard.hpp

```diff
@@ -171,12 +345,73 @@ class SFML_WINDOW_API Keyboard
     ////////////////////////////////////////////////////////////
     static bool isKeyPressed(Key key);
 
+    ////////////////////////////////////////////////////////////
+    /// \brief Check if a key is pressed
+    ///
+    /// \param code Scancode to check
+    ///
+    /// \return True if the physical key is pressed, false otherwise
+    ///
+    ////////////////////////////////////////////////////////////
+    static bool isKeyPressed(Scancode code);
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Localize a physical key to a logical one
+    ///
+    /// \param code Scancode to localize
+    ///
+    /// \return The key corresponding to the scancode under the current
+    ///         keyboard layout used by the operating system, or
+    ///         sf::Keyboard::Unknown when the scancode cannot be mapped
+    ///         to a Key.
+    ///
+    /// \see unlocalize
+    ///
+    ////////////////////////////////////////////////////////////
+    static Key localize(Scancode code);
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Identify the physical key corresponding to a logical one
+    ///
+    /// \param key Key to "unlocalize"
+    ///
+    /// \return The scancode corresponding to the key under the current
+    ///         keyboard layout used by the operating system, or
+    ///         sf::Keyboard::ScanUnknown when the key cannot be mapped
+    ///         to a sf::Keyboard::Scancode.
+    ///
+    /// \see localize
+    ///
+    ////////////////////////////////////////////////////////////
+    static Scancode unlocalize(Key key);
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Provide a string representation for a given scancode
+    ///
+    /// From a high level point of view, this conversion corresponds
+    /// somewhat to the string available through sf::Event::TextEvent
+    /// when the given physical key is pressed by the user, when no
+    /// modifiers are involved.
```

Comment:
"From a high level point of view" and "somewhat" -- what does this mean?
Does it correspond to `sf::Event::TextEvent` without modifiers or not?

Also, in that case I think `getDescription()` is not a good name. Maybe more along the lines of `getTextRepresentation()` or similar...

Reply:
The documentation text isn't as great, so I'll come up with something a bit more specific. Every OS does it a bit different. On Windows you get OS language specific translations, such as "UMSCHLT" for "Shift". On macOS and Linux we translate the UTF-8 value for letters and numbers, use some OS name resolving on Linux and have manual fallbacks on Linux and macOS.

Not sure if we should really unroll the discussion again on the naming, see here: https://github.com/SFML/SFML/pull/1235#discussion_r127936554

Description is quite fitting from my point of view, since it indicates that these aren't "technical" values and you shouldn't build logic on top, plus it's short and matches the purpose, to describe the key.
"Representation" kind be quite ambiguous in different context and makes it sound like, it's a technical value.

Reply:
Ah, good points. 

The idea that these are values meant to be displayed the user should not rely on their representation (e.g. for keys in maps) seems very important to me, maybe it should be mentioned in the doc of `getDescription()`.